src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -298,6 +299,40 @@
           (app req)))
       (app req))))
 
+;; for testing via with-redefs
+(defn get-sync-ver []
+  constants/pdb-sync-ver)
+
+(defn verify-sync-version
+  "Check that the x-pdb-sync-ver header of an incoming requests matches the
+   pdb-sync-ver locally. This header is only sent from pe-puppetdb sync requests
+   and a mismatch indicates that one pdb has upgraded in an incompatible way
+   before the other. In this case all sync requests will return 409s until both
+   sides have been upgraded and have matching sync versions. If the header is
+   present but improperly formatted returns a 400 error response."
+  [app]
+  (fn [{:keys [headers] :as req}]
```

Comment:
Not sure, but contemplated using :strs, i.e. `{{:strs [x-pdb-sync-ver]} :headers}` and and then could have something like `(some-> x-pdb-sync-ver Integer/parseInt)` in the `try` instead of the `when-let/get`.

Reply:
yep, that looks good. I'll update this